grails-app/services/org/broadinstitute/orsp/PermissionService.groovy

```diff
@@ -0,0 +1,50 @@
+package org.broadinstitute.orsp
+
+import grails.gorm.transactions.Transactional
+import groovy.util.logging.Slf4j
+
+@Transactional
+@Slf4j
+class PermissionService implements UserInfo{
+
+    UserService userService
+
+    // get issue's collaborators as a List<String>
+    def getIssueCollaborators(issue) {
+        def collaborators = issue.extraProperties.findAll ({ it.name == 'collaborator' }).collect { property -> property.value }
+        collaborators
+    }
+
+    // get issue's pms as a List<String>
+    def getIssuePMs(issue) {
+        def pms = issue.extraProperties.findAll ({ it.name == 'pm' }).collect { property -> property.value }
+        pms
+    }
+
+    // get issue's pis as a List<String>
+    def getIssuePIs(issue) {
+        def pis = issue.extraProperties.findAll ({ it.name == 'pi' }).collect { property -> property.value }
+        pis
+    }
+
+    // get issue's actors as a List<String>
+    def getIssueActors(issue) {
+        def actors = issue.extraProperties.findAll ({ it.name == 'actor' }).collect { property -> property.value }
+        actors
+    }
+
+    // verifies if logged user belogns to some user list ....
+    def issueIsForbidden(issue, userName, isAdmin, isReadOnlyAdmin) {
+
+        if (issue.reporter == userName
+                || getIssueCollaborators(issue).indexOf(userName) >= 0
+                || getIssuePMs(issue).indexOf(userName) >= 0
+                || getIssuePIs(issue).indexOf(userName) >= 0
+                || isAdmin
+                || isReadOnlyAdmin) {
+            return false
+        }
+        true
+    }
```

Comment:
This took me some time to grok as it's doing inverse logic. To make this easier to spot more quickly, can I suggest that we define a named variable to the result of the if and then return the negation of that? Something along the lines of:
```
        boolean userHasAccess = (issue.reporter == userName
                || getIssueCollaborators(issue).indexOf(userName) >= 0
                || getIssuePMs(issue).indexOf(userName) >= 0
                || getIssuePIs(issue).indexOf(userName) >= 0
                || isAdmin
                || isReadOnlyAdmin)
        !userHasAccess
```

Reply:
Alternatively, instead of checking for `forbidden-ness`, can we instead check for `allowed access` ? I think that would make the code a bit more readable.